### ISSUE TITLE
Implement cookie banner on frontend

### ISSUE THREAD

**Issue by Bento007** [chanzuckerberg/corpora-data-portal#332]
[UPDATE]
PR is out for review! Probably not worth estimating. Updated points from 3 -> 1

===============

The corpora data portal frontend must get users approval to use tracking cookies in the users browser. These cookies are used for google analytics.

This will require displaying a banner to inform users that tracking cookies are used.

The frontend implementation will match the implementation currently in [cellxgene](https://github.com/chanzuckerberg/cellxgene/blob/master/client/src/components/termsPrompt/index.js). @pdugan20 will update the design for it to gel better with the Corpora temp landing page.

**Comment by brianraymor** [comment]
What additional work is required to the existing implementation and design in Milestone 1?

![Screen Shot 2020-06-02 at 5.35.12 PM.png](https://images.zenhubusercontent.com/5b632709929b6b4421540ba4/eff45229-5cd3-40b7-90fd-b3fa9f795228)

**Comment by Bento007** [comment]
@brianraymor where does the above image come from?

**Comment by brianraymor** [comment]
If you clear your browser cache, navigate to cziscience.cellxgene.com, and click on a cellxgene link, you'll be offered that prompt. We've already worked with Legal related to the use of google analytics in the COVID-19 Corpora prototype.

**Comment by Bento007** [comment]
@brianraymor where did that image come from?

**Comment by mweiden** [comment]
Note the scaffold for this code is here https://github.com/chanzuckerberg/cellxgene/blob/master/client/src/components/termsPrompt/index.js Perhaps this will help implementation.

**Comment by maniarathi** [comment]
Chatted with @tihuan and @pdugan20 offline. This ticket will be split into two. The first part will be mimicking the functionality of the banner today. @pdugan20 will "corpora-ify" the banner to make it fit better with the current landing page design and @tihuan will implement as part of the landing page implementation.

The second part will involve sorting out the interaction of the cookies and AuthN which is dependent on BrianM finishing up his work first. 

So tl;dr: this ticket can be closed once the cookie banner is actually on the site.

**Comment by pdugan20** [comment]
Based on our earlier work, I think option 1 is preferable here:
https://www.figma.com/file/dvjDRK32DuCV0wQdfSYnsR/Cookie-Banner?node-id=0%3A1.

Having said that, since we are pivoting away from having a distinct brand from cellxgene, is it better to implement the original banner in the screenshot above?

**Comment by tihuan** [comment]
@pdugan20 Ooh that's a great question! Yeah consistency sounds good, so either cellxgene can update the banner using the new design or Data Portal can use the existing cellxgene screenshot above

From engineering's point of view, if we just wanna get the job done, copying cellxgene's design in the screenshot above is probably the easiest. But I do like the new design, since it looks much better 😆

**Comment by brianraymor** [comment]
I also like the new design. It's much cleaner and easier to read. 

Perhaps Colin and you could agree on a common approach. [Related code](https://github.com/chanzuckerberg/cellxgene/commits/c34a68304ec1c59d8adb55ff12403a020c1bf8d3/client/src/components/termsPrompt/index.js). cellxgene also added links to their [information menu](https://github.com/chanzuckerberg/cellxgene/pull/1300)

**Comment by tihuan** [comment]
@pdugan20 @colinmegill PTAL 😄 🙏 Thank you!

(I'm not blocked by the decision this week, but I should be able to work on this sometime next week. Thank you!)

**Comment by pdugan20** [comment]
I'll follow up with @colinmegill directly. Since the banner code is fairly straightforward and since we're just trying to stylistically match the cellxgene design rather than reference a single component, I think it's probably fine to use the new design here unless there are specific objections.

I'll ping the thread when we have confirmation.

**Comment by pdugan20** [comment]
@tihuan we're good to go with the design flagged above!

**Comment by tihuan** [comment]
Sweeeet! You rock, @pdugan20 !! New design it is 🤩 👏 ✨  Thanks a lot!
